Add CMCC compsets and one test for the coupled configuration in CMCC …

daniele-peano opened the pull request:
…framework

### Description of changes

Added a set of compset's aliases needed for the CMCC-CM development

Added an ERR automatic test for validating the coupled configuration in the CMCC-CM framework


### Specific notes

Contributors other than yourself, if any:

Fixes: [Github issue #s] And brief description of each issue.

User interface changes?: [ No/Yes ]
[ If yes, describe what changed, and steps taken to ensure backward compatibility ]

Testing performed (automated tests and/or manual tests):

daniele-peano committed:
Add CMCC compsets and one test for the coupled configuration in CMCC framework

mvertens commented on cime_config/testmods_dirs/allactive/def_cmcc_83l/shell_commands:
```diff
@@ -0,0 +1,17 @@
+./xmlchange RUN_STARTDATE=0001-01-01
```
For a fully coupled configuration - I would think that you wanted to put the ocean on their own pes. I think what you have hear is less than optimal. It would be good to do a PFS test under two load balancing scenarios and see if you have the right pe-layout for juno.

daniele-peano replied:
We performed some tests in the previous configuration, and having the ocean on its own pes did not increase the performance of the simulation since the atmosphere is the slowest component in this configuration.
By the way, I should have included this PES in the config_pes, and I should have removed the NTASKS changes from here.
Are the PFS tests a test option like ERR?

mvertens commented on cime_config/testmods_dirs/allactive/def_cmcc_83l/user_nl_cpl:
```diff
@@ -0,0 +1,19 @@
+!------------------------------------------------------------------------
```
You should not need to add an empty user_nl_cpl if you have not changed any of the settings.

daniele-peano replied:
You are right, I will delete it.

mvertens commented on cime_config/testmods_dirs/allactive/def_cmcc_83l/user_nl_nemo:
```diff
@@ -0,0 +1,23 @@
+!----------------------------------------------------------------------------------
```
Same comment as above. If nemo does not have any custom settings for this use case you should not include it under def_cmcc_831.

daniele-peano replied:
As above, I will delete the file.

daniele-peano committed:
Set the PES for CMCC coupled simulations and tests

daniele-peano committed:
remove user_nl_cpl and user_nl_nemo from def_cmcc_83l

daniele-peano committed:
Add performance test case category perf_cmcc and update the default PES on Juno

daniele-peano committed:
Update the Externals.cfg to account for:

- Add of Automatic test for CMCC configurations and Juno machine
- Add CMCC compsets
- Add WW3 unstructured grid definition
- Update of WW3 repository
